spec/models/rubric_criterion_spec.rb

```diff
@@ -249,6 +249,46 @@
       @levels = @criterion.levels
     end
 
+    context 'editing levels edits marks' do
+      before(:each) do
+        @marks = @criterion.marks
+        @result = create(:result, marking_state: Result::MARKING_STATES[:incomplete])
+        @marks.create(mark: 0, result: @result)
+        @marks.create(mark: 1, result: @result)
```

Comment:
@ashih2018 the problem is you're trying to create three different marks for the same criterion and result. You need to create three different results, and associate one mark to each one.

Also, you should not be making `@result` an instance variable, as it isn't used in the tests later on.